rpc/gnmi/decimal64-deprecation.md

```diff
@@ -0,0 +1,168 @@
+## Proposal to deprecate Decimal64 in gNMI
+
+**Updated**: April 28, 2022
+**Author**: Carl Lebsack
+
+### Proposal
+
+Deprecate and schedule removal of the Decimal64 message from gNMI and stipulate
+all Decimal64 values in the OpenConfig models be delivered as
+gNMI.TypedValue.double_val.
+
+### Why is Decimal64 present today?
+
+OpenConfig models are defined in [YANG](https://tools.ietf.org/html/rfc6020), a
+modeling language which does not have a native IEEE-754 compliant binary
+floating point type and has instead chosen to only
+[support](https://tools.ietf.org/html/rfc6020#section-9.3) Decimal64 for real
+number representation. Because of this choice, models developed within YANG,
+including those of OpenConfig, have selected Decimal64 for data values where
+fractional representation is necessary.
+
+### Why is Decimal64 problematic for gNMI?
+
+The primary problem is inefficiency which manifests in multiple ways by virtue
+of a misalignment of the natural encoding of the real data and the decimal
+encoding of the Decimal64 type. Before enumerating some of the inefficiencies in
+practical use, we’ll start with a brief discussion of the ultimate source of the
+data being modeled and the inappropriateness of Decimal64 for encoding.
+
+All values within the OpenConfig models using the YANG Decimal64 type are
```

Comment:
One note here is that values that are /actually/ floating point values (e.g., in IS-IS we have fields that are float32 because they are received in LSPs as float32) use a specific `ieeefloat32` `typedef` already.